electron/electron.js

```diff
@@ -53,3 +54,10 @@ app.on('activate', () => {
 
 // In this file you can include the rest of your app's specific main process
 // code. You can also put them in separate files and import them here.
+
+ipcMain.handle('run-extraction', async (event, fromDate, toDate, configFilepath, runLogFilepath, debug, allEntries) => {
+  const extractedData = await runExtraction(fromDate, toDate, configFilepath, runLogFilepath, debug, allEntries).then(
+    (value) => value,
```

Comment:
I think that the `.then` statement here isn't necessary. Because you're using the `await` keyword when calling `runExtraction` this will wait until the promise is resolved in `runExtraction` before the `extractedData` variable is assigned. This makes it so there isn't a promise that needs to wait for a `.then` here, so you should be safe to remove it.